core/src/main/java/org/apache/accumulo/core/fate/zookeeper/ZooCache.java

```diff
@@ -59,17 +63,29 @@ public interface ZooCacheWatcher extends Consumer<WatchedEvent> {}
 
   private static final Logger log = LoggerFactory.getLogger(ZooCache.class);
 
-  private final ZCacheWatcher watcher = new ZCacheWatcher();
-  private final Optional<ZooCacheWatcher> externalWatcher;
+  protected final TreeSet<String> watchedPaths = new TreeSet<>();
```

Comment:
This is modfied outside of the constructor so it needs to be made a synchronized set or a volatile pointer to an immutable set that gets recreated w/ each change.

Reply:
Implemented in 4da2f2c.